python/lsst/pipe/base/argumentParser.py

```diff
@@ -564,8 +564,10 @@ def __init__(self, pattern):
 
                     def write(self, str):
                         str = str.rstrip()
-                        if str and fnmatch.fnmatch(str, self._pattern):
-                            print str
+                        # Strip off doc string line(s) and cut off at "=" for string matching
+                        matchStr = str.split("\n")[-1].split("=")[0]
```

Comment:
Not your fault, but we shouldn't be using `str` as a variable name --- it clobbers the class name.